test/metabase/api/permissions_test.clj

```diff
@@ -100,6 +100,10 @@
         (is (= nil
                (get id->member :trashbird)))))
 
+    (testing "returns 404 for nonexistent id"
+      (is (= "API endpoint does not exist."
+             (mt/user-http-request :rasta :get 404 (format "permissions/group/%d" (:id 10000))))))
```

Comment:
Here you're using the `:id` keyword as a function, and calling it on an integer. Keywords as functions are usually used to look up values in maps, such as `(:a {:a :b}) -> :b`. But it so happens that when you call it on an integer it returns `nil`, which is probably not what you were intending here.

```
=> (format "permissions/group/%d" (:id 10000))
permissions/group/null
```
This is why the error message is `API endpoint does not exist` instead of `Not found`. It doesn't match any route since it's expecting an integer there.

You'll also need to change the test user to `:crowberto` to get it working once you've fixed that. This route is admin-only and crowberto is our admin test user. (The list of test users is [here](https://github.com/metabase/metabase/blob/master/test/metabase/test/data/users.clj#L34))